src/components/schedule-visualizer/colors.ts

```diff
@@ -41,6 +37,23 @@ export default class ColorManager {
     return color ? color : null;
   }
 
+  /**
+   * Gets a light color for the path. The hash is an async function that takes a while to give a response. Since our application should graph the path in real time, the hash was discarded and the path id was chosen to calculate its color. An attempt was made on robot-trajectory.tsx to get the colors with the hash by wrapping it in a useEffect but the update time was not fast enough.
+   * @param pathId : id of the path
+   */
+  static getPathColor(pathId: number): string {
```

Comment:
``` typescript
  async robotColor(name: string, model: string): Promise<string> {
    const key = this._robotKey(name, model);
    let color = this._robotColorCache[key];
    if (!color) {
      const modelHash = new Uint16Array(await _hash(model));
      const nameHash = new Uint16Array(await _hash(name));
      color = ColorManager._getLightColor(modelHash[0], nameHash[0]);
      this._robotColorCache[key] = color;
      this._robotColorCache[name] = color;
    }
    return color;
  }
```


Ok, I'm using the `robotColorFromCache` now, but the original function was using the `name` and `model` of the robot to build a key. In the trajectory case, we just have the `robot_name` and the `fleet_name`. What I did in the meanwhile it's to create a new key with just the name of the robot and with that, the trajectory has the same color as the Robot that as you said makes sense.

So, I need to ask some questions here:
* Is the name of the robot unique or just the key-value pair of robot + model are unique.
* Can I use the robot + model to create the color of the Robot and just use the robot_name as a key?. In case the robot names are unique of course.
* I think this is the less valid option and doesn't make a lot of sense but still worth asking, can we add to the robot model to the trajectory?

Reply:
> Is the name of the robot unique or just the key-value pair of robot + model are unique.

Yes, they should be unique.

> Can I use the robot + model to create the color of the Robot and just use the robot_name as a key?. In case the robot names are unique of course.

That makes sense, the names are unique so there shouldn't be any issues.

> I think this is the less valid option and doesn't make a lot of sense but still worth asking, can we add to the robot model to the trajectory?

There has been a bit of discussion to have a service to query robot information, it's not a priority so it probably won't be available in the current release of rmf. When we eventually implement that service, the `RobotState` will likely no longer contain the model, instead we will use that service to query it based on the robot name.

Reply:
the color of robots are generated by name + model and the cache key is the name of the robot